File: modules/global/src/com/haulmont/addon/datagen/entity/bool/BooleanPropertyGenerationStrategy.java
package com.haulmont.addon.datagen.entity.bool;

import com.haulmont.chile.core.datatypes.impl.EnumClass;

import javax.annotation.Nullable;


public enum BooleanPropertyGenerationStrategy implements EnumClass<String> {

    RANDOM("RANDOM"),
    MANUAL("MANUAL");

    private String id;

    BooleanPropertyGenerationStrategy(String value) {
        this.id = value;
    }

    public String getId() {
        return id;
    }

    @Nullable
    public static BooleanPropertyGenerationStrategy fromId(String id) {
        for (BooleanPropertyGenerationStrategy at : BooleanPropertyGenerationStrategy.values()) {
            if (at.getId().equals(id)) {
                return at;
            }
        }
        return null;
    }
}

File: modules/global/src/com/haulmont/addon/datagen/entity/BooleanPropertyGenerationStrategy.kt
package com.haulmont.addon.datagen.entity

import com.haulmont.chile.core.datatypes.impl.EnumClass

enum class BooleanPropertyGenerationStrategy(private val id: String) : EnumClass<String> {
    RANDOM("RANDOM"),
    MANUAL("MANUAL");

    override fun getId(): String {
        return id
    }

    companion object {

        @JvmStatic
        fun fromId(id: String): BooleanPropertyGenerationStrategy? {
            for (at in BooleanPropertyGenerationStrategy.values()) {
                if (at.getId() == id) {
                    return at
                }
            }
            return null
        }
    }
}
